bundle-core/src/main/java/net/discdd/client/bundletransmission/BundleTransmission.java

```diff
@@ -165,19 +167,29 @@ private BundleDTO generateNewBundle(String bundleId) throws RoutingExceptions.Cl
         List<ADU> adus = this.applicationDataManager.fetchADUsToSend(clientPaths.BUNDLE_SIZE_LIMIT, null);
         var routingData = clientRouting.bundleMetaData();
 
-        var baos = new ByteArrayOutputStream();
-        var ackedEncryptedBundleId = ackRecord == null ? null : ackRecord.getBundleId();
-        BundleUtils.createBundlePayloadForAdus(adus, routingData, ackedEncryptedBundleId, baos);
+        Path tmpPath = clientPaths.tosendDir.resolve("tmp");
```

Comment:
we don't need a tmp file. we need to convert this to use DDDJarFIle. let's sitdown and rewrite this on tuesday